blocks/monitoring/evaluators.py

```diff
@@ -172,7 +196,20 @@ class DatasetEvaluator(object):
 
     """
     def __init__(self, variables, updates=None):
-        self.buffer_ = AggregationBuffer(variables)
+        theano_variables = []
+        non_theano_variables = []
+        for variable in variables:
+            if isinstance(variable, MonitoredQuantity):
+                non_theano_variables += [variable]
+            else:
+                theano_variables += [variable]
+        self.theano_variables = theano_variables
+        self.non_theano_variables = non_theano_variables
+        variable_names = [v.name for v in theano_variables + non_theano_variables]
+        if len(variable_names) < len(theano_variables + non_theano_variables):
+            raise ValueError("variables should have different names")
+        self.theano_buffer_ = AggregationBuffer(theano_variables)
```

Comment:
I don't remember why there was underscore in the name `buffer_`, maybe there was a conflict with some python global function. Since the names are now longer, you can remove it.